tests/node-tests/basic-node-tests.js

```diff
@@ -39,8 +39,23 @@ fluid.require("test-module", require, "test-module");
 
 fluid.setLogging(true);
 
+// Set boolean for the  presence of the --tap command line option to output a
+// TAP report for consumption by testem when used as a custom launcher
+var shouldOutputTAP = process.argv.findIndex(function (argument) {
+    return argument.indexOf("--tap") > -1;
+}) > -1;
+
+// For accumulating TAP messages for parsing by testem custom launcher
+var tapOutput = "";
+
 QUnit.testDone(function (data) {
-    fluid.log("Test concluded - " + data.name + ": " + data.passed + " passed, " + data.failed + " failed");
+    var testMessage = "Test concluded - " + data.name + ": " + data.passed + " passed, " + data.failed + " failed";
+    fluid.log(testMessage);
+    if (shouldOutputTAP) {
```

Comment:
@amb26 agree that factoring makes the code cleaner & have implemented - the TAP generation is now in separate `QUnit.testDone` / `QUnit.done` callbacks.

I've linked FLUID-5744 and FLUID-5986 as related issues, and added a comment to FLUID-5744 pointing to this PR discussion.